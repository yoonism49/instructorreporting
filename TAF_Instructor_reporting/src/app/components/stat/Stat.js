// @flow weak
/* eslint consistent-return: 0 */

import React      from 'react';
import PropTypes  from 'prop-types';

const Stat = ({
  statFaIconName,
  statIconColor,
  statLabel,
  statIconSpin
}) => (
  <div className="stat">
    <div style={{textAlign: 'center'}}>
      <span className="fa-stack fa-lg fa-3x">
        <i
          className="fa fa-circle fa-stack-2x"
          style={{color: '#FFFFFF'}}>
        </i>
        <i
          className={`fa ${statFaIconName} ${statIconSpin ? 'fa-spin' : ''} fa-stack-1x fa-inverse`}
          style={{color: statIconColor}}>
        </i>
      </span>
    </div>
    <h5 className="stat-info">
      {statLabel}
    </h5>
  </div>
);

Stat.propTypes = {
  statFaIconName: PropTypes.string.isRequired,
  statIconColor:  PropTypes.string,
  statLabel:      PropTypes.string.isRequired,
  statIconSpin:   PropTypes.bool.isRequired
};

Stat.defaultProps = {
  statColor:    '4A4A4A',
  statIconSpin: false
};

export default Stat;
