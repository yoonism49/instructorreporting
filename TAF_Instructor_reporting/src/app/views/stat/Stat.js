// @flow weak

import React, {
  PureComponent
}                         from 'react';
import PropTypes          from 'prop-types';
import {
  AnimatedView,
  Panel,
  Stat as StatComponent
}                         from '../../components';
import Highlight          from 'react-highlight';


class Stat extends PureComponent {
  componentWillMount() {
    const { actions: { enterStat } } = this.props;
    enterStat();
  }

  componentWillUnmount() {
    const { actions: { leaveStat } } = this.props;
    leaveStat();
  }

  render() {
    const source = `
      // import
      import { Stat } from './_SOMEWHERE_/components';

      // in render():
      <div className="row">
        <div className="col-md-2">
          <Stat
            statFaIconName="fa-file-o"
            statIconColor="#fa8564"
            statLabel="999 Projects"
          />
        </div>
        <div className="col-md-2">
          <Stat
            statFaIconName="fa-paperclip"
            statIconColor="#45cf95"
            statLabel="999 Documents"
          />
        </div>
        <div className="col-md-2">
          <Stat
            statFaIconName="fa-envelope-o"
            statIconColor="#AC75F0"
            statLabel="999 Messages"
          />
        </div>
        <div className="col-md-2">
          <Stat
            statFaIconName="fa-check-square-o"
            statIconColor="#45cf95"
            statLabel="1000 Tasks"
          />
        </div>
        <div className="col-md-2">
          <Stat
            statFaIconName="fa-dollar"
            statIconColor="#AC75F0"
            statLabel="$99999 Earnings"
          />
        </div>
        <div className="col-md-2">
          <Stat
            statFaIconName="fa-refresh"
            statIconColor="#fa8564"
            statIconSpin={true}
            statLabel="Processing...."
          />
        </div>
      </div>
      `;

    return(
      <AnimatedView>
        {/* preview: */}
        <div className="row">
          <div className="col-xs-12">
            <Panel
              title="Stat"
              hasTitle={true}
              bodyBackGndColor={'#F4F5F6'}>

              <div className="row">
                <div className="col-md-2">
                  <StatComponent
                    statFaIconName="fa-file-o"
                    statIconColor="#fa8564"
                    statLabel="999 Projects"
                  />
                </div>
                <div className="col-md-2">
                  <StatComponent
                    statFaIconName="fa-paperclip"
                    statIconColor="#45cf95"
                    statLabel="999 Documents"
                  />
                </div>
                <div className="col-md-2">
                  <StatComponent
                    statFaIconName="fa-envelope-o"
                    statIconColor="#AC75F0"
                    statLabel="999 Messages"
                  />
                </div>
                <div className="col-md-2">
                  <StatComponent
                    statFaIconName="fa-check-square-o"
                    statIconColor="#45cf95"
                    statLabel="1000 Tasks"
                  />
                </div>
                <div className="col-md-2">
                  <StatComponent
                    statFaIconName="fa-dollar"
                    statIconColor="#AC75F0"
                    statLabel="$99999 Earnings"
                  />
                </div>
                <div className="col-md-2">
                  <StatComponent
                    statFaIconName="fa-refresh"
                    statIconColor="#fa8564"
                    statIconSpin={true}
                    statLabel="Processing...."
                  />
                </div>
              </div>

            </Panel>
          </div>
        </div>
        {/* source: */}
        <div className="row">
          <div className="col-xs-12">
            <Panel
              title="Source"
              hasTitle={true}>
              <Highlight className="javascript">
                {source}
              </Highlight>
            </Panel>
          </div>
        </div>
      </AnimatedView>
    );
  }
}

Stat.propTypes= {
  actions: PropTypes.shape({
    enterStat: PropTypes.func.isRequired,
    leaveStat: PropTypes.func.isRequired
  })
};

export default Stat;
