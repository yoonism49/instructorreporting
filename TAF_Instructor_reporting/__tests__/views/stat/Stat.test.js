
// @flow weak

import React            from 'react';
import Stat             from '../../../src/app/views/stat/Stat';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import {
  shallow
}                       from 'enzyme';
import { 
  MemoryRouter
}                       from 'react-router';

// react-highlight uses findDOMNode: 
// -> jest will throw errors (no way to counter that...)
jest.mock('react-highlight');


describe('Stat view', () => {
  it('renders as expected', () => {
    const props = {
      actions: {
        enterStat: () => {},
        leaveStat: () => {}
      }
    };
    const component = renderer.create(
      <div>
        <MemoryRouter>
          <Stat {...props} />
        </MemoryRouter>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('triggers enterStat on mount', () => {
    const mockenterStat = jest.fn();
    const mockProps = {
      actions: {
        enterStat: mockenterStat,
        leaveStat: () => {}
      }
    };
    /* eslint-disable no-unused-vars */
    const wrapper = shallow(
      <Stat {...mockProps} />
    );
    expect(mockenterStat.mock.calls.length).toBe(1);
  });

  it('triggers leaveStat on unMount', () => {
    const mockleaveStat = jest.fn();
    const mockProps = {
      actions: {
        enterStat: () => {},
        leaveStat: mockleaveStat
      }
    };
    const wrapper = shallow(
      <Stat {...mockProps} />
    );
    wrapper.unmount();
    expect(mockleaveStat.mock.calls.length).toBe(1);
  });
});
