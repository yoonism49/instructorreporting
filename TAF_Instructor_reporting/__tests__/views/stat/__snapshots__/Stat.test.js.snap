// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat view renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Stat
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#F4F5F6",
              }
            }
          >
            <div
              className="row"
            >
              <div
                className="col-md-2"
              >
                <div
                  className="stat"
                >
                  <div
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    <span
                      className="fa-stack fa-lg fa-3x"
                    >
                      <i
                        className="fa fa-circle fa-stack-2x"
                        style={
                          Object {
                            "color": "#FFFFFF",
                          }
                        }
                      />
                      <i
                        className="fa fa-file-o  fa-stack-1x fa-inverse"
                        style={
                          Object {
                            "color": "#fa8564",
                          }
                        }
                      />
                    </span>
                  </div>
                  <h5
                    className="stat-info"
                  >
                    999 Projects
                  </h5>
                </div>
              </div>
              <div
                className="col-md-2"
              >
                <div
                  className="stat"
                >
                  <div
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    <span
                      className="fa-stack fa-lg fa-3x"
                    >
                      <i
                        className="fa fa-circle fa-stack-2x"
                        style={
                          Object {
                            "color": "#FFFFFF",
                          }
                        }
                      />
                      <i
                        className="fa fa-paperclip  fa-stack-1x fa-inverse"
                        style={
                          Object {
                            "color": "#45cf95",
                          }
                        }
                      />
                    </span>
                  </div>
                  <h5
                    className="stat-info"
                  >
                    999 Documents
                  </h5>
                </div>
              </div>
              <div
                className="col-md-2"
              >
                <div
                  className="stat"
                >
                  <div
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    <span
                      className="fa-stack fa-lg fa-3x"
                    >
                      <i
                        className="fa fa-circle fa-stack-2x"
                        style={
                          Object {
                            "color": "#FFFFFF",
                          }
                        }
                      />
                      <i
                        className="fa fa-envelope-o  fa-stack-1x fa-inverse"
                        style={
                          Object {
                            "color": "#AC75F0",
                          }
                        }
                      />
                    </span>
                  </div>
                  <h5
                    className="stat-info"
                  >
                    999 Messages
                  </h5>
                </div>
              </div>
              <div
                className="col-md-2"
              >
                <div
                  className="stat"
                >
                  <div
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    <span
                      className="fa-stack fa-lg fa-3x"
                    >
                      <i
                        className="fa fa-circle fa-stack-2x"
                        style={
                          Object {
                            "color": "#FFFFFF",
                          }
                        }
                      />
                      <i
                        className="fa fa-check-square-o  fa-stack-1x fa-inverse"
                        style={
                          Object {
                            "color": "#45cf95",
                          }
                        }
                      />
                    </span>
                  </div>
                  <h5
                    className="stat-info"
                  >
                    1000 Tasks
                  </h5>
                </div>
              </div>
              <div
                className="col-md-2"
              >
                <div
                  className="stat"
                >
                  <div
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    <span
                      className="fa-stack fa-lg fa-3x"
                    >
                      <i
                        className="fa fa-circle fa-stack-2x"
                        style={
                          Object {
                            "color": "#FFFFFF",
                          }
                        }
                      />
                      <i
                        className="fa fa-dollar  fa-stack-1x fa-inverse"
                        style={
                          Object {
                            "color": "#AC75F0",
                          }
                        }
                      />
                    </span>
                  </div>
                  <h5
                    className="stat-info"
                  >
                    $99999 Earnings
                  </h5>
                </div>
              </div>
              <div
                className="col-md-2"
              >
                <div
                  className="stat"
                >
                  <div
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    <span
                      className="fa-stack fa-lg fa-3x"
                    >
                      <i
                        className="fa fa-circle fa-stack-2x"
                        style={
                          Object {
                            "color": "#FFFFFF",
                          }
                        }
                      />
                      <i
                        className="fa fa-refresh fa-spin fa-stack-1x fa-inverse"
                        style={
                          Object {
                            "color": "#fa8564",
                          }
                        }
                      />
                    </span>
                  </div>
                  <h5
                    className="stat-info"
                  >
                    Processing....
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Source
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          />
        </section>
      </div>
    </div>
  </section>
</div>
`;
